comid/integregisters.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package comid
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/veraison/swid"
+)
+
+// IRegisterIndex is the interface to hold register index
+type IRegisterIndex interface{}
+
+// IntegReg holds the Integrity Registers
+type IntegReg struct {
```

Comment:
The choices in naming are very confusing: there are 5 different contractions for the same concept:
* IntegRegister
* IntegReg
* integregisters
* IRegisterIndex
* Register (which could be just `m` (map) or `v` (val) or `val`)

There is no good reason for shortening `IntegrityRegister`.  On the contrary, there are very good reasons to make sure the code can be related unambiguously to the spec.

Let's use `IntegrityRegister` consistently (also the file names).

Reply:
Renamed everywhere to IntegrityRegisters